backend/src/models/video.go

```diff
@@ -17,6 +17,7 @@ type Video struct {
 	VisibilityStatus      bool              `json:"visibility_status" gorm:"not null;default:false"`
 	ThumbnailUrl          string            `json:"thumbnail_url" gorm:"size:255"`
 	OpenContentProviderID uint              `json:"open_content_provider_id" gorm:"not null"`
+	IsFavorited           bool              `json:"is_favorited"`
```

Comment:
We can't add these `is_favorited` fields to the models themselves because gorm will try to insert them and it throws an error. I end up making another struct that embeds the model, and adds that field.
```go
type VideoResponse struct {
models.Video
IsFavorited bool `json:"is_favorited"`
}```

Reply:
Made the correction. I had a feeling that is why you used the LibraryResponse. Since I wasn't receiving any gorm errors I added it here to the model directly.

Reply:
its weird cuz I have seen it throw gorm errors, and other times it doesn't complain. I even tried adding an explicit `gorm:"-"` tag and it was throwing errors.